### PR TITLE
Align naming in Scenes cluster XML with the spec.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
REVIEW NOTE: Only the first commit contains manual changes. The rest is code-generated.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/scene.xml [hunk 1]
```diff
@@ -128,36 +128,36 @@ limitations under the License.
...
     
     <command source="server" code="0x01" name="ViewSceneResponse" optional="false" disableDefaultResponse="true">
       <description>
         The command is generated in response to a received unicast ViewScene command 
       </description>
       <arg name="Status" type="ENUM8"/>
-      <arg name="GroupId" type="group_id"/>
-      <arg name="SceneId" type="INT8U"/>
+      <arg name="GroupID" type="group_id"/>
+      <arg name="SceneID" type="INT8U"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/scene.xml:160]: should SceneName below have a length=16 constraint? Or does that not matter?

File: src/app/zap-templates/zcl/data-model/chip/scene.xml [hunk 2]
```diff
@@ -204,17 +204,17 @@ limitations under the License.
...
     </command>
 
     <command source="server" code="0x41" name="EnhancedViewSceneResponse" optional="true" disableDefaultResponse="true">
       <description>
         The command is generated in response to a received unicast EnhancedViewScene command
       </description>
       <arg name="Status" type="ENUM8"/>
-      <arg name="GroupId" type="group_Id"/>
-      <arg name="SceneId" type="INT8U"/>
+      <arg name="GroupID" type="group_Id"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/scene.xml:216]: some types are `group_id` and some are `group_Id`. Should we normalize casing on types as well?

File: src/app/zap-templates/zcl/data-model/chip/scene.xml [hunk 3]
```diff
@@ -225,13 +225,13 @@ limitations under the License.
...
-      <arg name="GroupIdFrom" type="group_Id"/>
-      <arg name="SceneIdFrom" type="INT8U"/>
+      <arg name="GroupIdentifierFrom" type="group_Id"/>
+      <arg name="SceneIdentifierFrom" type="INT8U"/>
     </command>
   </cluster>
 
   <bitmap name="SceneFeatures" type="BITMAP32">
-    <cluster code="0x0006" />
+    <cluster code="0x0005" />
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/scene.xml:234]: Nit: `Scene Names` below has a space in my copy of the spec.

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/scene.xml:234]: Yes, we have decided we don't want spaces in identifiers and I will be filing spec issues to remove those in the spec.

File: src/app/zap-templates/zcl/data-model/chip/scene.xml (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/scene.xml:216]: We probably should, as a followup; right now ZAP treats type names as kinda-case-insensitive-except-if-it's-a-named-type or something.  So changing the casing there does not affect codegen and we can do it any time easily.

File: src/app/zap-templates/zcl/data-model/chip/scene.xml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/scene.xml:160]: We do nothing with constraints for command fields right now.  So does not matter in practice.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**